src/P/Function.hs

```diff
@@ -1,12 +1,26 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 module P.Function (
     applyN
+  -- * Functions for development/debuggung only
+  -- | Cannot be used in production code, but might be useful
+  -- during development or debugging
+  , undefined
+  , error
```

Comment:
if you don't export it by default though, isn't someone more likely to import it from elsewhere where it doesn't have warnings? (I was at least, I didn't think to check whether P had development submodules)